order.h

```diff
@@ -101,134 +136,76 @@ using OrderSurface = struct OrderSurface {
 
 
 class MarketOrder: public Order {
-    MarketOrder(TickerSymbol ticker, Price price, Quantity quantity, Side side)
-    : Order(ticker, OrderType::MARKET, price, quantity, side) {}
-
-    [[nodiscard]] bool canMatch() {
```

Comment:
Good to remove, matching should not be happening at the Order level, matcher decides this. OFC lot of context would be missing if we try to do it at this level, and things would become to interdependent... should be layered and separate